src/dataaggregator/aggregator/policyPoller.go

```diff
@@ -0,0 +1,77 @@
+package aggregator
+
+import (
+	"code.cloudfoundry.org/lager"
+	"dataaggregator/db"
+	"dataaggregator/policy"
+	"time"
+)
+
+type PolicyPoller struct {
+	logger     lager.Logger
+	interval   int
+	database   db.DB
+	tick       *time.Ticker
+	doneChan   chan bool
+	PolicyMap  map[string]*policy.PolicyJson
+	TriggerMap map[string]*policy.Trigger
+}
+
+func NewPolicyPoller(logger lager.Logger, interval int, database db.DB) *PolicyPoller {
+	return &PolicyPoller{
+		logger:     logger,
+		interval:   interval,
+		database:   database,
+		doneChan:   make(chan bool),
+		PolicyMap:  make(map[string]*policy.PolicyJson),
+		TriggerMap: make(map[string]*policy.Trigger),
+	}
+}
+
+func (p *PolicyPoller) Start() {
+	p.tick = time.NewTicker(time.Duration(p.interval) * time.Second)
+	go p.startPolicyRetrieve()
+	p.logger.Info("policy-poller-started", lager.Data{"interval": p.interval})
+}
+
+func (p *PolicyPoller) Stop() {
+	if p.tick != nil {
+		p.tick.Stop()
+		close(p.doneChan)
+	}
+	p.logger.Info("policy-poller-stopped")
+}
+func (p *PolicyPoller) startPolicyRetrieve() {
+	for {
+		p.retrievePolicy()
+		select {
+		case <-p.doneChan:
+			return
+		case <-p.tick.C:
+		}
+	}
+}
+
+func (p *PolicyPoller) retrievePolicy() {
+	policies, err := p.database.RetrievePolicies()
+	if err != nil {
+		p.logger.Error("refresh-app", err)
+		return
+	}
+	p.logger.Info("policy count", lager.Data{"count": len(policies)})
+	for _, policyRow := range policies {
+		appId := policyRow.AppId
+		storedPolicyJson, ok := p.PolicyMap[policyRow.AppId]
+		if ok {
```

Comment:
This is highly redundant. Just determine if things need to be updated and have one block.